src/plugins/persistence/couch/CouchChangesFeed.js

```diff
@@ -33,80 +35,35 @@
-                } while (connected);
+                self.listenForChanges(event.data.url);
             }
         };
 
         port.start();
-
     };
 
-    self.onerror = function () {
-        //do nothing
-        console.log('Error on feed');
+    self.onerror = function (error) {
+        console.error('Error on feed', error);
     };
 
-    self.listenForChanges = async function (url, body, port) {
-        connected = true;
-        let error = false;
-        // feed=continuous maintains an indefinitely open connection with a keep-alive of HEARTBEAT milliseconds until this client closes the connection
-        // style=main_only returns only the current winning revision of the document
-
-        console.log('Opening changes feed connection.');
-        const response = await fetch(url, {
-            method: 'POST',
-            headers: {
-                "Content-Type": 'application/json'
-            },
-            signal,
-            body
+    self.onmessage = function (event) {
+        const objectChanges = JSON.parse(event.data);
+        connections.forEach(function (connection) {
+            connection.postMessage({
+                objectChanges
+            });
         });
+    };
 
-        let reader;
-
-        if (response.body === undefined) {
-            error = true;
-        } else {
-            reader = response.body.getReader();
-        }
-
-        while (!error) {
-            const {done, value} = await reader.read();
-            //done is true when we lose connection with the provider
-            if (done) {
-                error = true;
-            }
-
-            if (value) {
-                let chunk = new Uint8Array(value.length);
-                chunk.set(value, 0);
-                const decodedChunk = new TextDecoder("utf-8").decode(chunk).split('\n');
-                console.log('Received chunk');
-                if (decodedChunk.length && decodedChunk[decodedChunk.length - 1] === '') {
-                    decodedChunk.forEach((doc, index) => {
-                        try {
-                            if (doc) {
-                                const objectChanges = JSON.parse(doc);
-                                connections.forEach(function (connection) {
-                                    connection.postMessage({
-                                        objectChanges
-                                    });
-                                });
-                            }
-                        } catch (decodeError) {
-                            //do nothing;
-                            console.log(decodeError);
-                        }
-                    });
-                }
-            }
+    self.listenForChanges = function (url) {
+        console.debug('⇿ Opening CouchDB change feed connection ⇿');
 
-        }
+        couchEventSource = new EventSource(url, { withCredentials: false });
```

Comment:
What's the reason for setting `withCredentials: false`?

Reply:
I was experimenting with passing CORS credentials:
https://developer.mozilla.org/en-US/docs/Web/API/EventSource/withCredentials
`withCredentials: false` is the default, so we can omit this.